### ISSUE TITLE
Support for AdaptDL

### ISSUE THREAD

**Issue by pietrolesci** [Lightning-AI/lightning#15119]
Reporting from the `idea-pool` channel on slack, as discussed with @carmocca.

--- 

Hi there,

On the way to solve a OOM problem with dynamic batch sizes based on sequence length, I have just discovered [AdaptDL](https://adaptdl.readthedocs.io/en/latest/adaptdl-pytorch.html). Might be an interesting library to support.

Some core features offered by AdaptDL are:

- Elastically schedule distributed DL training jobs in shared clusters.
- Cost-aware resource auto-scaling in cloud computing environments (e.g. AWS).
- Automatic batch size and learning rate scaling for distributed training.

**Comment by carmocca** [comment]
Hi! Thanks for the request.

Note however that we would only be interested in integrating the training tricks they provide such as batch size and learning rate scaling features.

We already support easy and elastic scaling of cloud jobs via our [Lightning.ai](Lightning.ai) platform. You can find ready-to-use apps such as the [PL app](https://lightning.ai/app/WYU6CrDIAS-PyTorch%20Lightning%20App) to launch your PL scripts, or the [demo app](https://lightning.ai/app/AU3WoWwdAP-Train%20%26%20Demo%20PyTorch%20Lightning) to get started.
